analytics/analytics.go

```diff
@@ -5,91 +5,121 @@ import (
+}
 
-	anonymizedUsageGroup := AnonymizedUsageGroupModel{}
-	for _, stepRunResult := range stepRunResults {
-		anonymizedUsageData := AnonymizedUsageModel{
-			ID:      stepRunResult.StepInfo.ID,
-			Version: stepRunResult.StepInfo.Version,
-			RunTime: stepRunResult.RunTime.Seconds(),
-			Error:   stepRunResult.Status != 0,
-			CI:      configs.IsCIMode,
-		}
+func stepStatus(i int) string {
+	if status, ok := map[int]string{
+		models.StepRunStatusCodeFailed:           "failed",
+		models.StepRunStatusCodeSuccess:          "success",
+		models.StepRunStatusCodeSkipped:          "skipped",
+		models.StepRunStatusCodeFailedSkippable:  "failed_skippable",
+		models.StepRunStatusCodeSkippedWithRunIf: "skipped_with_runif",
+	}[i]; ok {
+		return status
+	}
+	return "unknown"
+}
 
-		anonymizedUsageGroup.Steps = append(anonymizedUsageGroup.Steps, anonymizedUsageData)
+// SendAnonymizedAnalytics ...
+func SendAnonymizedAnalytics(buildRunResults models.BuildRunResultsModel) error {
+	var (
+		runtime       time.Duration
+		stepAnalytics []StepAnalytics
+	)
+	for _, stepResult := range buildRunResults.OrderedResults() {
+		stepAnalytics, runtime = append(stepAnalytics, StepAnalytics{
+			StepID:    stepResult.StepInfo.ID,
+			Status:    stepStatus(stepResult.Status),
+			Runtime:   stepResult.RunTime,
+			StartTime: stepResult.StartTime,
+		}), runtime+stepResult.RunTime
 	}
 
-	data, err := json.Marshal(anonymizedUsageGroup)
-	if err != nil {
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(BuildAnalytics{
+		Runtime:    runtime,
+		StartTime:  buildRunResults.StartTime,
+		Platform:   buildRunResults.ProjectType,
+		StackID:    os.Getenv(stackIDEnvKey),
+		AppSlug:    os.Getenv(appSlugEnvKey),
+		BuildSlug:  os.Getenv(buildSlugEnvKey),
+		Status:     buildStatus(buildRunResults.IsBuildFailed()),
+		CLIVersion: os.Getenv(plugins.PluginInputBitriseVersionKey),
+	}); err != nil {
 		return err
 	}
 
-	req, err := http.NewRequest("POST", analyticsURL, bytes.NewBuffer(data))
+	req, err := http.NewRequest("POST", analyticsBaseURL+"/metrics", &body)
 	if err != nil {
-		return fmt.Errorf("failed to create request with usage data (%s), error: %s", string(data), err)
+		return fmt.Errorf("failed to create request with usage data (%s), error: %s", body.String(), err)
 	}
-
 	req.Header.Set("Content-Type", "application/json")
 
-	timeout := time.Duration(10 * time.Second)
 	client := http.Client{
-		Timeout: timeout,
+		Timeout: time.Duration(10 * time.Second),
 	}
 	resp, err := client.Do(req)
 	if err != nil {
-		return fmt.Errorf("failed to perform request with usage data (%s), error: %s", string(data), err)
+		return fmt.Errorf("failed to perform request with usage data (%s), error: %s", body.String(), err)
 	}
-
 	defer func() {
 		if err := resp.Body.Close(); err != nil {
 			log.Errorf("failed to close response body, error: %#v", err)
 		}
 	}()
 
 	if resp.StatusCode < 200 || resp.StatusCode > 210 {
```

Comment:
Are you sure about this status `code 210`? Is it a valid status code?

Reply:
Yeah, we've done the backend server. We know what we can expect. Other than that: 👍